prog/dftb+/prg_dftb/main.F90

```diff
@@ -0,0 +1,5057 @@
+            & iCellVec, cellVec, kPoint, kWeight, orb, species, solver, tRealHS, tSpinSharedEf,&
+            & tSpinOrbit, tDualSpinOrbit, tFillKSep, tFixEf, tMulliken, iDistribFn, tempElec, nEl,&
+            & Ef, energy, eigen, filling, rhoPrim, Eband, TS, E0, iHam, xi, orbitalL, HSqrReal,&
+            & SSqrReal, iRhoPrim, HSqrCplx, SSqrCplx, rhoSqrReal, storeEigvecsReal,&
+            & storeEigvecsCplx)
+
+        if (tWriteBandDat) then
+          call writeBandOut(fdBand, bandOut, eigen, filling, kWeight)
+        end if
+
+        if (tMulliken) then
+          call getMullikenPopulation(rhoPrim, over, orb, neighborList, nNeighbor, img2CentCell,&
+              & iPair, qOutput, iRhoPrim=iRhoPrim, qBlock=qBlockOut, qiBlock=qiBlockOut)
+        end if
+
+        ! For non-dual spin-orbit orbitalL is determined during getDensity() call above
+        if (tDualSpinOrbit) then
+          call getL(orbitalL, qiBlockOut, orb, species)
+        end if
+
+        ! Note: if XLBOMD is active, potential created with input charges is needed later,
+        ! therefore it should not be overwritten here.
+        if (tSCC .and. .not. tXlbomd) then
+          call resetInternalPotentials(tDualSpinOrbit, xi, orb, species, potential)
+          call getChargePerShell(qOutput, orb, species, chargePerShell)
+          call addChargePotentials(qOutput, q0, chargePerShell, orb, species, species0,&
+              & neighborList, img2CentCell, spinW, thirdOrd, potential)
+          call addBlockChargePotentials(qBlockOut, qiBlockOut, tDftbU, tImHam, species, orb,&
+              & nDftbUFunc, UJ, nUJ, iUJ, niUJ, potential)
+          potential%intBlock = potential%intBlock + potential%extBlock
+        end if
+
+        call getEnergies(qOutput, q0, chargePerShell, species, tEField, tScc, tXlbomd, tDftbU,&
+            & tDualSpinOrbit, rhoPrim, H0, orb, neighborList, nNeighbor, img2CentCell, iPair,&
+            & cellVol, pressure, TS, potential, energy, thirdOrd, qBlockOut, qiBlockOut,&
+            & nDftbUFunc, UJ, nUJ, iUJ, niUJ, xi)
+
+        tStopScc = hasStopFile(fStopScc)
+
+        if (tScc) then
+          call getNextInputCharges(pChrgMixer, qOutput, qOutRed, orb, nIneqOrb, iEqOrbitals,&
+              & iGeoStep, iSccIter, minSccIter, maxSccIter, sccTol, tStopScc, tDftbU, tReadChrg,&
+              & qInput, qInpRed, sccErrorQ, tConverged, qBlockOut, iEqBlockDftbU, qBlockIn,&
+              & qiBlockOut, iEqBlockDftbULS, species0, nUJ, iUJ, niUJ, qiBlockIn)
+          call getSccInfo(iSccIter, energy%Eelec, Eold, diffElec)
+          call printSccInfo(tDftbU, iSccIter, energy%Eelec, diffElec, sccErrorQ)
+          tWriteSccRestart = needsSccRestartWriting(restartFreq, iGeoStep, iSccIter, minSccIter,&
+              & maxSccIter, tMd, tGeoOpt, tDerivs, tConverged, tReadChrg, tStopScc)
+          if (tWriteSccRestart) then
+            call writeRestart(fChargeIn, orb, qInput, qBlockIn, qiBlockIn)
```

Comment:
Added explicit file descriptor passing (we should at some point dispose this in favour of `newunit` in the  `open` statement.